src/Generator.h

```diff
@@ -2304,10 +2304,18 @@ class GeneratorStub : public NamesInterface {
         return get_first_output().realize(std::forward<Args>(args)..., get_target());
     }
 
-    template<typename Dst>
-    void realize(Dst dst) {
+    template<typename Realization>
+    void realize(Realization r) {
         check_scheduled("realize");
-        get_first_output().realize(dst, get_target());
+        get_first_output().realize(r, get_target());
+    }
+
+    // Note that the Realization here is expected to have the appropriate number-and-type
+    // of buffers to realize *all* of this Generator's Outputs (not just the first one).
```

Comment:
The interesting question here then is what the other variants of realize() in GeneratorStub should do. Perhaps the simplest and most consistent thing would be to have all of them realize all of the Outputs. (Users could still realize on a per-Output basis by using the explicit members.)